tests/e2e/fine_grained_access/path_filtering_tests.py

```diff
@@ -0,0 +1,432 @@
+    )
+
+    assert len(source_destination_path) == 0
+
+
+def test_dfs_denied_destination():
+    admin_connection = common.connect(username="admin", password="test")
+    user_connnection = common.connect(username="user", password="test")
+    common.execute_and_fetch_all(admin_connection.cursor(), "REVOKE LABELS * FROM user;")
+    common.execute_and_fetch_all(admin_connection.cursor(), "REVOKE EDGE_TYPES * FROM user;")
+    common.execute_and_fetch_all(
+        admin_connection.cursor(), "GRANT READ ON LABELS :label0, :label1, :label2, :label3 TO user;"
+    )
+    common.execute_and_fetch_all(admin_connection.cursor(), "GRANT READ ON EDGE_TYPES * TO user;")
+    common.execute_and_fetch_all(admin_connection.cursor(), "DENY READ ON LABELS :label4 TO user;")
+
+    source_destination_path = common.execute_and_fetch_all(
+        user_connnection.cursor(), "MATCH p=(n:label0)-[*]->(m:label4) RETURN p;"
+    )
+
+    assert len(source_destination_path) == 0
+
+
+def test_dfs_denied_label_1():
+    admin_connection = common.connect(username="admin", password="test")
+    user_connnection = common.connect(username="user", password="test")
+    common.execute_and_fetch_all(admin_connection.cursor(), "REVOKE LABELS * FROM user;")
+    common.execute_and_fetch_all(admin_connection.cursor(), "REVOKE EDGE_TYPES * FROM user;")
+    common.execute_and_fetch_all(
+        admin_connection.cursor(), "GRANT READ ON LABELS :label0, :label2, :label3, :label4 TO user;"
+    )
+    common.execute_and_fetch_all(admin_connection.cursor(), "DENY READ ON LABELS :label1 TO user;")
+    common.execute_and_fetch_all(admin_connection.cursor(), "GRANT READ ON EDGE_TYPES * TO user;")
+    source_destination_paths = common.execute_and_fetch_all(
+        user_connnection.cursor(), "MATCH p=(n:label0)-[*]->(m:label4) RETURN nodes(p);"
+    )
+
+    assert len(source_destination_paths) == 6
+    assert not any("label1" in node.labels for path in source_destination_paths for node in path[0])
+
+
+def test_dfs_denied_edge_type_3():
+    admin_connection = common.connect(username="admin", password="test")
+    user_connnection = common.connect(username="user", password="test")
+    common.execute_and_fetch_all(admin_connection.cursor(), "REVOKE LABELS * FROM user;")
+    common.execute_and_fetch_all(admin_connection.cursor(), "REVOKE EDGE_TYPES * FROM user;")
+
+    common.execute_and_fetch_all(admin_connection.cursor(), "GRANT READ ON LABELS * TO user;")
+    common.execute_and_fetch_all(
+        admin_connection.cursor(), "GRANT READ ON EDGE_TYPES :edge_type_1, :edge_type_2, :edge_type_4 TO user;"
+    )
+    common.execute_and_fetch_all(admin_connection.cursor(), "DENY READ ON EDGE_TYPES :edge_type_3 TO user;")
+
+    source_destination_paths = common.execute_and_fetch_all(
+        user_connnection.cursor(), "MATCH p=(n:label0)-[r *]->(m:label4) RETURN r;"
+    )
+
+    assert len(source_destination_paths) == 6
```

Comment:
same here and every path below that you tested. sorry but it's bulletproof like that